### PR TITLE
SLE15-SP2 Raspberry Pi updates

### PR DESCRIPTION
**Pull request by afaerber**
### Description
Initial round of text updates for Raspberry Pi Quick Start and Deployment Guide.
jsc#SLE-7277, jsc#SLE-7276

### Checklist
* Check all items that apply.

*Are backports required?*

- [ ] To maintenance/SLE15SP1
- [ ] To maintenance/SLE15SP0
- [ ] To maintenance/SLE12SP5
- [ ] To maintenance/SLE12SP4
- [ ] To maintenance/SLE12SP3

### CODE REVIEW

**Comment by afaerber** [comment]
Larger changes to the Deployment Guide are in progress still.

File: xml/art_sles_rpi_quick.xml [hunk 1]
```diff
@@ -51,8 +51,8 @@
   <para>
    To be able to use &productnamearch; on the &rpi;, a 64-bit Arm* compatible
```
Comment (lproven) [inline xml/art_sles_rpi_quick.xml]: Shouldn't we be using an entity for Arm here?

File: xml/art_sles_rpi_quick.xml [hunk 2]
```diff
@@ -51,8 +51,8 @@
...
    &rpi;* is required. &productnamearch; &productnumber; is tested to work
-   on &rpi; 3 Model B and Model B+ boards
-   as well as on &rpi; Compute Module 3 on MyPi Industrial IoT Integrator Board.
+   on &rpi; 3 Model B, Model B+, Model A+ and 4 Model B boards
```
Comment (lproven) [inline xml/art_sles_rpi_quick.xml]: I suggest listing the variants in alphabetical order.

File: xml/art_sles_rpi_quick.xml [hunk 3]
```diff
@@ -211,12 +211,12 @@
    <itemizedlist>
     <listitem>
      <para>
-      Raspberry Pi 3 Model B or Raspberry Pi 3 Model B+
+      &rpi; 3 Model B or &rpi; 3 Model B+ or 3 Model A+ or &rpi; 4 Model B
```
Comment (lproven) [inline xml/art_sles_rpi_quick.xml]: I suggest:

"Raspberry Pi 3 Model A+, B or B+, or Raspberry Pi 4 Model B". 

I find out-of-order model names hard to follow.

File: xml/art_sles_rpi_quick.xml [hunk 4]
```diff
@@ -395,8 +395,8 @@
 
   <para>
    &appliancename; is distributed as XZ-compressed image file for
-   microSD cards. This section will guide you through the process of preparing
-   the card to the first boot. If you already have a microSD card containing
+   &microsd; cards. This section will guide you through the process of preparing
+   the card to the first boot. If you already have a &microsd; card containing
```
Comment (lproven) [inline xml/art_sles_rpi_quick.xml]:
```suggestion
   the card for the first boot. If you already have a &microsd; card containing
```

File: xml/art_sles_rpi_quick.xml [hunk 5]
```diff
@@ -434,10 +434,10 @@
     <title>Writing the Image to the Card using <command>dd</command></title>
     <para>
      This command decompresses the image
-     <literal>SLES15-SP1-JeOS.aarch64-15.1-RaspberryPi-GM.raw.xz</literal> to the SD
+     <literal>SLES15-SP2-JeOS.aarch64-15.2-RaspberryPi-GM.raw.xz</literal> to the SD
```
Comment (lproven) [inline xml/art_sles_rpi_quick.xml]:
```suggestion
     <literal>SLES15-SP2-JeOS.aarch64-15.2-RaspberryPi-GM.raw.xz</literal> to the &microsd;
```

File: xml/art_sles_rpi_quick.xml (hunk 4 above)
Comment (afaerber) [reply xml/art_sles_rpi_quick.xml]: True, but unrelated to this commit.

File: xml/art_sles_rpi_quick.xml (hunk 5 above)
Comment (afaerber) [reply xml/art_sles_rpi_quick.xml]: We have 11 more occurrences of "SD card", which is not strictly wrong - the reason for &microsd; was the inconsistent spelling MicroSD vs. microSD. We could probably clean that up with more time on our hands.

File: xml/art_sles_rpi_quick.xml (hunk 1 above)
Comment (afaerber) [reply xml/art_sles_rpi_quick.xml]: Let's consider such cleanups as follow-ups once the upcoming bulk of contents is in.

File: xml/art_sles_rpi_quick.xml (hunk 3 above)
Comment (afaerber) [reply xml/art_sles_rpi_quick.xml]: Sounds okay, but it would really help me if you commented on my cleanly separated commits rather on the overall diff, so that I know where to change things.

File: xml/art_sles_rpi_quick.xml (hunk 1 above)
Comment (afaerber) [reply xml/art_sles_rpi_quick.xml]: I've found a couple more unused entities even.

**Comment by afaerber** [comment]
The huge update is now pushed, along with most requested cleanups. Please consider merging.

File: xml/art_sles_rpi_quick.xml [hunk 6]
```diff
@@ -893,8 +892,8 @@ Password:
    </note>
    <para>
     <emphasis role="strong">After you obtained a registration code from a
```
Comment (cjschroder) [inline xml/art_sles_rpi_quick.xml]: "After you obtained" should be "After you obtain".

File: xml/deployment_prep_aarch64_raspi.xml [hunk 7]
```diff
@@ -29,162 +29,266 @@
...
+    must share the total 480&nbsp;Mbps bandwidth of the USB 2.0 controller.
+    This will limit performance, and could significantly impact network performance.
+   </para>
+   <para>
+    This limitation does not apply to the &rpi; 4.
+   </para>
+  </note>
+  <para>
+   Newer models of &rpi; 3 with BCM2837 B0 silicon (silver instead of black chip),
+   including &rpi; 3 Model B+ and Compute Module 3+, allow to boot from USB-connected
```
Comment (cjschroder) [inline xml/deployment_prep_aarch64_raspi.xml]: "allow to boot" should be "allow booting"

File: xml/deployment_prep_aarch64_raspi.xml [hunk 8]
```diff
@@ -29,162 +29,266 @@
...
+   Installation from Network
   </title>
   <para>
-   Due to the hardware's lack of on-board firmware, network-booting the &rpi;
-   using PXE is more complex than with more conventional computers.
+   Due to the hardware's lack of on-board firmware (<xref linkend="sec-aarch64-deprpi-boot"/>),
+   network-booting the &rpi; using PXE is more complex than with more conventional computers.
   </para>
   <para>
    The process of setting up a PXE boot server for &x86; and &arm; is described
```
Comment (cjschroder) [inline xml/deployment_prep_aarch64_raspi.xml]: The URL needs a description: "...is described in `<citetitle>Network boot your Raspberry Pi</citetitle>`

File: xml/deployment_prep_aarch64_raspi.xml [hunk 9]
```diff
@@ -215,7 +320,7 @@
       section of the &sls; Release Notes:
```
Comment (cjschroder) [inline xml/deployment_prep_aarch64_raspi.xml]: "consult the relevant section of the &sls; Release Notes" should be "consult the `<citetitle>ARM 64-Bit-Specific Features & Fixes (AArch64)</citetitle>` section of the &sls; Release Notes"

File: xml/deployment_prep_aarch64_raspi.xml (hunk 8 above)
Comment (afaerber) [reply xml/deployment_prep_aarch64_raspi.xml]: Why? I tested the PDF to look fine. And note that I renamed that section.

Comment (afaerber) [reply xml/deployment_prep_aarch64_raspi.xml]: Ah, note that the link is **not** to `-net` but to `-boot`. It's about the preceding "on-board firmware", not about the "network-booting", which is this very section.

File: xml/art_sles_rpi_quick.xml (hunk 6 above)
Comment (afaerber) [reply xml/art_sles_rpi_quick.xml]: Not my text, so shouldn't block the update please.
Would "After you have obtained..." work, too?

Comment (cjschroder) [reply xml/art_sles_rpi_quick.xml]: yes that works too

File: xml/deployment_prep_aarch64_raspi.xml (hunk 8 above)
Comment (cjschroder) [reply xml/deployment_prep_aarch64_raspi.xml]: URLs always need a description of some kind so readers know when they're in the right place. URLs change and sometime Web browsers don't render deep links correctly. We use text descriptions rather than URL labels because URL labels render incorrectly in PDFs.

**Comment by cjschroder** [comment]
I'm fine with merging this now, and then I will proofread and write a separate commit with corrections. @afaerber does that work for you? @lproven I am off til Monday-- if Andreas is OK with merging this as-is will you merge? thanks!

File: xml/deployment_prep_aarch64_raspi.xml (hunk 8 above)
Comment (afaerber) [reply xml/deployment_prep_aarch64_raspi.xml]: Did you maybe comment on the wrong link? There is no URL here! It's an internal section reference that shows fine in the PDF as: `Section 3.6.1, "Boot Process"`

File: xml/deployment_prep_aarch64_raspi.xml [hunk 10]
```diff
@@ -101,105 +242,53 @@
...
-  </para>
-
  </sect2>
- <sect2>
+
+ <sect2 xml:id="sec-aarch64-deprpi-net">
   <title>
-   Network Booting the &rpi;
+   Installation from Network
   </title>
```
Comment (afaerber) [inline xml/deployment_prep_aarch64_raspi.xml]: This pre-existing multi-line title is causing whitespace artifacts, will collapse to one-line as the others.

File: xml/deployment_prep_aarch64_raspi.xml (hunk 9 above)
Comment (afaerber) [reply xml/deployment_prep_aarch64_raspi.xml]: I'll use the newly merged "Boot and Driver Enablement for Raspberry Pi" instead.

File: xml/deployment_prep_aarch64_raspi.xml (hunk 7 above)
Comment (afaerber) [reply xml/deployment_prep_aarch64_raspi.xml]: @cjschroder Can you please explain why? I've changed a second instance of "allow to boot" as requested above, but there is also "allow to deploy" - is "allow to" generally disliked, or is it "to boot" that is the culprit?

File: xml/deployment_prep_aarch64_raspi.xml [hunk 11]
```diff
@@ -27,6 +27,11 @@
     &rpi; 3 Model B+
    </para>
   </listitem>
+  <listitem>
+   <para>
+    &rpi; 3 Model A+
+   </para>
+  </listitem>
```
Comment (afaerber) [inline xml/deployment_prep_aarch64_raspi.xml]: This one is still ordered after B+, will change.

**Comment by afaerber** [comment]
> I'm fine with merging this now, and then I will proofread and write a separate commit with corrections. @afaerber does that work for you? @lproven I am off til Monday-- if Andreas is OK with merging this as-is will you merge? thanks!

@lproven @cjschroder From my view, with the provided updates, this is ready for merging, yes.
Having this accessible in HTML form will aid PM and engineers to review the new contents.

Please let me review any further correction pull requests for this. I'll still be working on minor updates for the Quick Start guide (some of which we may want to backport to SP1, too) as well as some more contents additions for the Deployment guide (Device Tree, U-Boot, network boot, limitations of RPi4, appliance building).

Btw the Quick Start guide talks of "respective sections" of Deployment and Administration guides for network setup (4.3)  - can you help to turn those into actual links?

I also wonder what your plan is for the Quick Start vs. Deployment split - should we be moving the image deployment instructions to the Deployment guide and reference it from the Quick Start guide rather than the opposite direction? Or just for Linux and leave the Windows/Mac parts in Quick Start? Compare the new USB Boot Mode section in Deployment guide referencing parts of the procedure vs. Appliance deployment pointing to the Quick Start only.

**Comment by afaerber** [comment]
Converted one more occurrence of the SCC URL to an entity.

File: xml/deployment_prep_aarch64_raspi.xml (hunk 7 above)
Comment (lproven) [reply xml/deployment_prep_aarch64_raspi.xml]: > is "allow to" generally disliked

Well, yes. The word following the transitive verb "allow" is its subject: the noun which is being permitted. So a gerund ("booting," "running") is acceptable, but an infinitive verb is not. We would need the wordier "allows the computer to boot", for example.

Comment (afaerber) [reply xml/deployment_prep_aarch64_raspi.xml]: Many thanks for explaining. Addressed in #564.

**Comment by afaerber** [comment]
@lproven With your merge you have squashed all my tidy, meticulously crafted commits into one single ugly commit 300639bef0ae8b91746eb248d54f1ae74bb006e6 - please do not do that again!

**Comment by afaerber** [comment]
Also, commit 9b2c45bd90a039b640a03ef3c1ec8b51caf38710 was pushed without any notification to me nor review, disregarding [my above comment](https://github.com/SUSE/doc-sle/pull/563#issuecomment-598704123) and causing further rebase problems.
